### PR TITLE
Adds typing to components module and improve docstring.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

## Summary
- Adds typing to components module
- Improve docstrings

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

File: maliput_sim/core/components.py [hunk 1]
```diff
@@ -27,19 +27,42 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+class Entity: pass
+class EntityComponentManager: pass
+class SimulationState: pass
```
Comment (francocipollone) [inline maliput_sim/core/components.py:32]: This forward declaration doesn't cope well with type checks (via mypy).

Given that for example the method `def set_entity(self, entity: Entity):` will actually expecting a `maliput_sim.core.components.Entity` when I want it to expect a `maliput_sim.core.ecm.Entity`
Of course, adding an import of that `Entity` in this module will cause a cyclic dependency.

Comment (francocipollone) [reply maliput_sim/core/components.py:32]: I still need to resolve this, by the way, @agalbachicar do you have any experience with this case?

Comment (agalbachicar) [reply maliput_sim/core/components.py:32]: I don't have experience with this either. However, I've read the documentation of [type hints](https://peps.python.org/pep-0484/#forward-references) and it seems like this is dealt with a string literal type hint.

Comment (francocipollone) [reply maliput_sim/core/components.py:32]: Oh that's useful, I missed that. Thanks!

File: maliput_sim/core/components.py [hunk 2]
```diff
@@ -27,19 +27,48 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+from typing import List
+
+import maliput_sim
```
Comment (francocipollone) [inline maliput_sim/core/components.py:32]: If here I do 
`from maliput_sim.core.ecm import Entity` instead and in the 43 line I do
`    def set_entity(self, entity: 'Entity'):` then I get a cyclincg dependency

```
ImportError: cannot import name 'Component' from partially initialized module 'maliput_sim.core.components' (most likely due to a circular import) (/home/franco/maliput_ws/install/maliput_sim/lib/python3.8/site-packages/maliput_sim/core/components.py)
```

**Comment by francocipollone** [comment]
Ready to review @agalbachicar 
I will add tests for the `components` module in a follow up PR

File: maliput_sim/core/components.py [hunk 3]
```diff
@@ -27,19 +27,48 @@
...
+        self._entity = None
+
+    def set_entity(self, entity: 'maliput_sim.core.ecm.Entity'):
+        """
+        Set the entity that this component belongs to.
+
+        Args:
+            entity: The entity that this component belongs to.
+
+        """
```
Comment (agalbachicar) [inline maliput_sim/core/components.py:50]:
```suggestion
        """
```

File: maliput_sim/core/components.py [hunk 4]
```diff
@@ -27,19 +27,48 @@
...
+            entity: The entity that this component belongs to.
+            ecm: The entity component manager.
+        """
         pass
 
     def get_state(self):
+        """
+        Get the current state of the component.
+        """
         return None
```
Comment (agalbachicar) [inline maliput_sim/core/components.py:72]: Isn't it better if we do `pass` here?

**Review by agalbachicar — APPROVED** [review]
LGTM

File: maliput_sim/core/components.py (hunk 4 above)
Comment (francocipollone) [reply maliput_sim/core/components.py:72]: Indeed. Done

File: maliput_sim/core/components.py (hunk 3 above)
Comment (francocipollone) [reply maliput_sim/core/components.py:50]: Done

**Pull request merged**